api/app/Services/SocialAccount/Google.php

```diff
@@ -37,15 +47,33 @@ public function service(string $service)
         return new $classname($this->client);
     }
 
-    public function auth(string $code = null)
+    public function auth(string $code = null, string $state = null)
     {
         if (!$code) {
             return $this->createAuthUrl();
         } else {
             $this->authenticate($code);
+            $user = $this->decodeUser($state);
+
             $token = $this->getAccessToken();
 
-            return '';
+            if (!$token) {
+                throw new GoogleOauthException();
+            }
+
+            $socialAccount = $this->socialAccountRepository->findByProvider(
+                SocialAccount::GOOGLE_SERVICE_ID,
+                $user->userId
+            );
+
+            $socialAccount->account_id = $user->userId;
+            $socialAccount->token = $token;
+            $socialAccount->refresh_token = $token['refresh_token'];
+            $socialAccount->expires_in = '2020-08-27 22:57:59';
+
+            $socialAccount->save();
+
+            return 'Success';
```

Comment:
We should not return such values. The execution can be treated as a success if there were not any exceptions.